confgenerator/confgenerator.go

```diff
@@ -200,53 +204,84 @@ type fileReceiverFactory struct {
 	ExcludePaths []string
 }
 
-func extractReceiverFactories(receivers map[string]*receiver) (map[string]*fileReceiverFactory, map[string]*syslogReceiverFactory, error) {
+type winlogReceiverFactory struct {
+	Channels []string
+}
+
+func extractReceiverFactories(receivers map[string]*receiver) (map[string]*fileReceiverFactory, map[string]*syslogReceiverFactory, map[string]*winlogReceiverFactory, error) {
 	fileReceiverFactories := map[string]*fileReceiverFactory{}
 	syslogReceiverFactories := map[string]*syslogReceiverFactory{}
+	winlogReceiverFactories := map[string]*winlogReceiverFactory{}
 	for n, r := range receivers {
 		if strings.HasPrefix(n, "lib:") {
-			return nil, nil, fmt.Errorf(`receiver id prefix 'lib:' is reserved for pre-defined receivers. Receiver ID %q is not allowed.`, n)
+			return nil, nil, nil, fmt.Errorf(`receiver id prefix 'lib:' is reserved for pre-defined receivers. Receiver ID %q is not allowed.`, n)
 		}
 		switch r.Type {
 		case "files":
```

Comment:
[Not directly related to this PR] We don't have to deal with this in this PR.

It may be more convenient to do an `inclusive` check than the `exclusive` check here for future convenience. With `inclusive` checks, we do not have to touch every case when adding new params.

Reply:
ACK